src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,986 @@
+                // foreach (var a in ...)
+                // ...
+                // {
+                //      var x = a;
+                //      dosomething(x); 
+                //  }
+                var statements = GenerateStatements(
+                    expression => AddToBlockTop(SyntaxFactory.LocalDeclarationStatement(
+                        SyntaxFactory.VariableDeclaration(
+                            forEachStatement.Type,
+                            SyntaxFactory.SingletonSeparatedList(
+                                SyntaxFactory.VariableDeclarator(
+                                    forEachStatement.Identifier,
+                                    argumentList: null,
+                                    SyntaxFactory.EqualsValueClause(expression))))),
+                                        forEachStatement.Statement).WithAdditionalAnnotations(Formatter.Annotation), queryExpressionProcessingInfo);
+                return new DocumentUpdateInfo(forEachStatement, statements);
+            }
+
+            private DocumentUpdateInfo ConvertIfInToForeachWithoutExtraVariableDeclaration(
+                ForEachStatementSyntax forEachStatement,
+                QueryExpressionProcessingInfo queryExpressionProcessingInfo)
+            {
+                // before 
+                //  foreach (var a in from a in b where a > 5 select a) 
+                //  { 
+                //      dosomething(a); 
+                //  }
+                // after 
+                //  foreach (var a in b)
+                //  {
+                //      if (a > 5)
+                //      {
+                //          dosomething(a); 
+                //      }
+                //  }
+                var statements = GenerateStatements(
+                    expression => forEachStatement.Statement.WithAdditionalAnnotations(Formatter.Annotation),
+                    queryExpressionProcessingInfo);
+                return new DocumentUpdateInfo(forEachStatement, statements);
+            }
+
+            private bool TryConvertIfInReturnStatement(
+                ReturnStatementSyntax returnStatement,
+                QueryExpressionProcessingInfo queryExpressionProcessingInfo,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                // The conversion requires yield return which cannot be added to lambdas and anonymous method declarations.
+                if (IsWithinImmediateLambdaOrAnonymousMethod(returnStatement))
+                {
+                    documentUpdateInfo = default;
+                    return false;
+                }
+
+                var memberDeclarationNode = FindParentMemberDeclarationNode(returnStatement);
+                if (memberDeclarationNode == null)
+                {
+                    documentUpdateInfo = default;
+                    return false;
+                }
+
+                var declaredSymbol = _semanticModel.GetDeclaredSymbol(memberDeclarationNode);
```

Comment:
>var declaredSymbol = _semanticModel.GetDeclaredSymbol(memberDeclarationNode); [](start = 16, length = 77)

We already found the symbol in ```FindParentMemberDeclarationNode```, consider returning it from there as an out parameter.